##                           TAREA 2

#### 1. Supongamos que x es un vector numérico. Explica en detalle, como las siguientes expresiones son evaluadas y que valores toman.

EXPRESION      | ¿QUE HACE?
:-------|------:
sum(!is.na(x)) | En primera instancia "!is.na(x)" devuelve TRUE o FALSE para cada elemento evaluado en el vector,TRUE si no es un valor perdido (NA) y FALSE si lo es,luego "sum" identifica estos resultados como ceros y unos,para luego sumarlos. En otras palabras esta expresion calcula el numero de elementos que no son NA.
c(x,x[-(1:length(x))])|En primera instancia "x[-(1:length(x))]" devuelve un vector,el cual resulta de quitarle a x los elementos que se encuentren en las posiciones comprendidas desde el 1 hasta length(x),pero vemos que todos los elementos de x cumplen esta condicion,por lo tanto al final se devolvera un vector sin elementos.En segunda instancia la expresion c(x,y) indexa dos vectores,por lo tanto para nuestro ejemplo,el vector "y" sera un vector sin elementos y la indexacion dara como resultado el mismo vector x.
x[length(x) + 1]/length(x)|"x[length(x) + 1]" devuelve el elemento de x que encuentre en la posicion "length(x) + 1" pero no existe ningun valor de x en ese posicion,ya que x solo poee length(x) elementos,luego devolvera el valor NA (elemento perdido o no encontrado).La division posterior simplemente seguira siendo NA.
sum(x > mean(x))|Esta expresion nos da como resultado el numero de elementos que que esten por encima de la media de x.

In [32]:
#Aqui un ejemplo:
x=c(3,6,9,4,2)
sum(!is.na(x))
c(x,x[-(1:length(x))])
x[length(x) + 1]/length(x)
sum(x >mean(x))

[1] 5

[1] 3 6 9 4 2

[1] NA

[1] 2

#### 2. Crear una función vectorizada en R, que calcule el sinh.

In [34]:
sinh <- function(x) 
{
    (exp(x)-exp(-x))/2
}
 # Ejemplo:
   sinh(2)

[1] 3.62686

#### 3. Crea un vector de los valores de (e^x)cos ( x ) en x = 3, 3.1, 3.2, . . . , 6.

In [38]:
# creamos x = 3, 3.1, 3.2, . . . , 6.
x <- seq(3,6,by=0.1)
y <- exp(x)*cos(x) 
y


[1] -19.884531 -22.178753 -24.490697 -26.773182 -28.969238 -31.011186
 [7] -32.819775 -34.303360 -35.357194 -35.862834 -35.687732 -34.685042
[13] -32.693695 -29.538816 -25.032529 -18.975233 -11.157417  -1.362099
[19]  10.632038  25.046705  42.099201  61.996630  84.929067 111.061586
[25] 140.525075 173.405776 209.733494 249.468441 292.486707 338.564378
[31] 387.360340

#### 4. Usa la función paste para crear el siguiente vector de caracteres de longitud 30:
* ("vector 1", "vector 2", . . . , "vector 30"). Se notar el espacio entre vector y el número.
* ("fn1", "fn2", . . . , "fn30"). En este caso no hay espacio entre fn y el número.

In [45]:
paste(c(rep("vector ", 30)),1:30)

[1] "vector  1"  "vector  2"  "vector  3"  "vector  4"  "vector  5" 
 [6] "vector  6"  "vector  7"  "vector  8"  "vector  9"  "vector  10"
[11] "vector  11" "vector  12" "vector  13" "vector  14" "vector  15"
[16] "vector  16" "vector  17" "vector  18" "vector  19" "vector  20"
[21] "vector  21" "vector  22" "vector  23" "vector  24" "vector  25"
[26] "vector  26" "vector  27" "vector  28" "vector  29" "vector  30"

In [46]:
paste(c(rep("fn", 30)),1:30)

[1] "fn 1"  "fn 2"  "fn 3"  "fn 4"  "fn 5"  "fn 6"  "fn 7"  "fn 8"  "fn 9" 
[10] "fn 10" "fn 11" "fn 12" "fn 13" "fn 14" "fn 15" "fn 16" "fn 17" "fn 18"
[19] "fn 19" "fn 20" "fn 21" "fn 22" "fn 23" "fn 24" "fn 25" "fn 26" "fn 27"
[28] "fn 28" "fn 29" "fn 30"

#### 5. ¿Qué retorna DF <- D(expression(cos(x)/sin(x)), "x") . Si ejecutamos el comando x <- pi/4 ,que retorna eval(DF) ?.

#### 6. ¿Puedes explicar los dos siguientes resultados? 
* x <- c(0,7,8)
* x[0.9999999999999999]
* numeric(0)
* x[0.99999999999999999]
* 0

x[a],siendo "a" un numero decimal,devuelve el elemento que se encuentra en la posicion numericamente igual al maximo entero de a.
luego el numero 0.9999999999999999 es tomado y reconocido como tal,cuyo maximo entero es 0 por ende x[0.9999999999999999] es equivalente a x[0] y como no existe un numero en esa posicion no imprimira nada,mientas que 0.99999999999999999 es indistinguible con el numero 1 debido al grado de precision del programa(no se distingue a dos numeros cuya diferencia sea menor a 10^-16),luego x[0.9999999999999999] es equivalente a x[1],imprimiendose el valor en la posicion 1,en este ejemplo se imprimira 0.

#### 7. Usando rep() y seq() , crea los siguientes vectores
0 0 0 0 0 1 1 1 1 1 2 2 2 2 2 3 3 3 3 3 4 4 4 4 4

1 2 3 4 5 1 2 3 4 5 1 2 3 4 5 1 2 3 4 5 1 2 3  4 5

In [82]:
x<-seq(0,4,by=1)
y<-c()
for(i in 1:5)y<-c(y,rep(x[i],5))
    y

[1] 0 0 0 0 0 1 1 1 1 1 2 2 2 2 2 3 3 3 3 3 4 4 4 4 4

In [70]:
rep(seq(1,5,by=1),5)

[1] 1 2 3 4 5 1 2 3 4 5 1 2 3 4 5 1 2 3 4 5 1 2 3 4 5

####  8.Usando la función cumprod o otra relacionada, calcula:

![Sin titulo](sumatoria.png)

In [83]:
sum(cumprod(seq(2,38,by=2)/seq(3,39,by=2)))+1

[1] 6.976346

#### 9. En primer lugar, lanzamos una moneda con dos posibles resultados: o bien cae cara o sello.A continuación, tiramos un dado con 6 posibles resultados: 1, 2, 3, 4, 5 y 6. Genera todos los resultados posibles de esas acciones.

In [88]:
library(sets)
A= set("C","S")
B= set("1","2","3","4","5","6")
A*B

{("C", "1"), ("C", "2"), ("C", "3"), ("C", "4"), ("C", "5"), ("C",
 "6"), ("S", "1"), ("S", "2"), ("S", "3"), ("S", "4"), ("S", "5"),
 ("S", "6")}

#### 10. Tienes un conjunto de colores para elegir:
> colores <- c("rojo", "azul", "verde", "blanco", "negro", "amarillo")
Ahora debes elegir 3 colores y no puedes "elegir el mismo color más de una vez. Enumera todas las
combinaciones posibles.

In [112]:
colores <- c("rojo", "azul", "verde", "blanco", "negro", "amarillo")
combn(colores,3)
#Numero de combinaciones posibles:
dim(combn(colores,3))[2]


[,1]  [,2]   [,3]  [,4]     [,5]   [,6]  [,7]     [,8]   [,9]     [,10]   
[1,] rojo  rojo   rojo  rojo     rojo   rojo  rojo     rojo   rojo     rojo    
[2,] azul  azul   azul  azul     verde  verde verde    blanco blanco   negro   
[3,] verde blanco negro amarillo blanco negro amarillo negro  amarillo amarillo
     [,11]  [,12] [,13]    [,14]  [,15]    [,16]    [,17]  [,18]    [,19]   
[1,] azul   azul  azul     azul   azul     azul     verde  verde    verde   
[2,] verde  verde verde    blanco blanco   negro    blanco blanco   negro   
[3,] blanco negro amarillo negro  amarillo amarillo negro  amarillo amarillo
     [,20]   
[1,] blanco  
[2,] negro   
[3,] amarillo

[1] 20

#### 11. Tienes el mismo conjunto de colores y puedes elegir 1, 2 o 3. Haz una lista de todas las opciones posibles.

In [113]:
colores <- c("rojo", "azul", "verde", "blanco", "negro", "amarillo")
combn(colores,1)
combn(colores,2)
combn(colores,3)

[,1] [,2] [,3]  [,4]   [,5]  [,6]    
[1,] rojo azul verde blanco negro amarillo

[,1] [,2]  [,3]   [,4]  [,5]     [,6]  [,7]   [,8]  [,9]     [,10]  [,11]
[1,] rojo rojo  rojo   rojo  rojo     azul  azul   azul  azul     verde  verde
[2,] azul verde blanco negro amarillo verde blanco negro amarillo blanco negro
     [,12]    [,13]  [,14]    [,15]   
[1,] verde    blanco blanco   negro   
[2,] amarillo negro  amarillo amarillo

[,1]  [,2]   [,3]  [,4]     [,5]   [,6]  [,7]     [,8]   [,9]     [,10]   
[1,] rojo  rojo   rojo  rojo     rojo   rojo  rojo     rojo   rojo     rojo    
[2,] azul  azul   azul  azul     verde  verde verde    blanco blanco   negro   
[3,] verde blanco negro amarillo blanco negro amarillo negro  amarillo amarillo
     [,11]  [,12] [,13]    [,14]  [,15]    [,16]    [,17]  [,18]    [,19]   
[1,] azul   azul  azul     azul   azul     azul     verde  verde    verde   
[2,] verde  verde verde    blanco blanco   negro    blanco blanco   negro   
[3,] blanco negro amarillo negro  amarillo amarillo negro  amarillo amarillo
     [,20]   
[1,] blanco  
[2,] negro   
[3,] amarillo

#### 12.

![Sin titulo](12a.png)

![Sin titulo](12b.png)

#### 13. 
(a) ¿Cuál es la probabilidad de que una persona sea menor o igual a 1.90 m? Usa pnorm .


(b) ? Cuál es la probabilidad de que una persona sea más alta o igual a 1.60 m?. Usa pnor

#### 14. ¿Cuál es la probabilidad de que una persona espere menos de 10 minutos?. Utiliza pexp.

#### 15. Ejecuta las siguientes líneas que crean dos vectores de enteros aleatorios que se eligen con reemplazo de los enteros 0, 1, . . . , 999 . Ambos vectores tienen una longitud de 250.

In [115]:
set.seed(50)
xVec <- sample(0:999, 250, replace=T)
yVec <- sample(0:999, 250, replace=T)
xVec
yVec

[1] 708 437 200 767 513  44 699 646  42 107 390 269 640  77 277 676 835 364
 [19]  74 168 616 193 710 842 309 650 577 257 324 368 358 408 437 618 222 627
 [37] 121 701 373 458 363 836 278  93  55 700 954 458 713 803 996 765 639 299
 [55] 358 425 715 525 511 266 578 655 197 585 129  38 724  61 136 944 507 995
 [73] 661  74 967 148 657 956 652 956 543  17 339 469 544  19   1 680 537 645
 [91] 691 688 828 760  48 294  69 807 311 668 505 964 632   8  24 862  10 614
[109] 840 353 878  72 193 113  82 322  91 789 444 986 624  18 537 554 515 460
[127] 263  42  76 256 359 189 807 457  99 274 543 324 176 477 541 160 260 174
[145]  48 415 707 625 530 407 216 224 395 977 828 461 148 293 660  38 137 224
[163] 852 743 683 545 353 371 866 452 811 768 339 203 478  49  20 880 480 996
[181] 894 357 900 603 667 787 972 457 467 324 928 109 365 987 572 280 113 702
[199] 963 405  63 621 517 446 533 190 638 275 865 435 501 669 124  14 920 308
[217]  84 523   5 863 860 120 206 399  29 256 678  59 497 188 127 258 376 171
[235] 781 870 110 957 285 382  34 403 631 197 179 545 123 760 238 178

[1] 709 871 315 517 621 930 437 948 157 783 878 471 671  91 415 860 273 768
 [19] 581 381  47 347 229   4 279 411 698 974 554 279 216 855 813 776 218 721
 [37] 538 332  31 460 532 917 985  95 705 248 247 884 317 840  94 288  43 575
 [55] 687 174 213 957 955 786 938 428 930 101 330 641 615 988 500 285  28 881
 [73] 106 329 398 414 542 570 881 997 221 488 117 299 484 823 428 791 133  50
 [91] 246  72 520 643 779 693 845 296 441 553 815 752 465  18 766  87 635 257
[109] 993 368 919 116 224 686 473 151 512 635 613 660 310 419 800 428 743 282
[127] 965  44 330  19 743 615 489 615 194 803 948 760 604 193 409 800 772 133
[145] 175 593 184 516 287 863 902 195 220 689 309  14 881 941 924 593 693 280
[163] 835 632 225 398 872 876 358 187 211 850 961 681 791 947 117 915 222 712
[181] 665 921 798 167 421 268 866 503 828 942 589 521 320 424  13 482 498 509
[199] 216   0  78 488 841 645 681 827  83 273 421 277 884  67 890 970 400  10
[217] 469 290 632 717 529 426 127 846  49 952 609  99 284 824 598 695  63 293
[235] 325 295 675 777 813 557 792 580 783  72 611 853 738 345 668 791